src/main/java/life/qbic/projectbrowser/components/LevelComponent.java

```diff
@@ -489,6 +495,7 @@ public void itemClick(ItemClickEvent event) {
           samples.put(dataset.getCode(), dataset.getSampleIdentifierOrNull().split("/")[2]);
         }
 
+        // TODO we might want to make these datasets unique here or before, so they don't get listed multiple times
```

Comment:
I'd argue that theoretically we should throw an exception if a dataset is listed multiple times since that is not expected behaviour. 
For now you could think about converting it to linkedhashset since that keeps the ordering intact but allows for set filtering https://docs.oracle.com/javase/6/docs/api/java/util/LinkedHashSet.html. 
Another alternative would be to use the distinct() method within a stream:
https://howtodoinjava.com/java8/java-stream-distinct-examples/